From 53dd4c6aa6115cc397c5a0573050cb26687f7612 Mon Sep 17 00:00:00 2001
From: Tomas Glozar <tglozar@gmail.com>
Date: Fri, 14 Nov 2025 19:37:51 +0100
Subject: [PATCH 91/94] ia64: Switch from GENERIC_IRQ_LEGACY

GENERIC_IRQ_LEGACY was removed in commit 3c716487936a ("genirq: Remove
GENERIC_IRQ_LEGACY").

Move ia64 to use the new interface.

Signed-off-by: Tomas Glozar <tglozar@gmail.com>
---
 arch/ia64/kernel/iosapic.c  | 2 +-
 arch/ia64/kernel/irq_ia64.c | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/ia64/kernel/iosapic.c b/arch/ia64/kernel/iosapic.c
index f37e4402c635..cad414d34b5e 100644
--- a/arch/ia64/kernel/iosapic.c
+++ b/arch/ia64/kernel/iosapic.c
@@ -734,7 +734,7 @@ iosapic_register_intr (unsigned int gsi,
 		rte = find_rte(irq, gsi);
 		if(iosapic_intr_info[irq].count == 0) {
 			assign_irq_vector(irq);
-			irq_init_desc(irq);
+			irq_free_descs(irq, 1);
 		} else if (rte->refcnt != NO_REF_RTE) {
 			rte->refcnt++;
 			goto unlock_iosapic_lock;
diff --git a/arch/ia64/kernel/irq_ia64.c b/arch/ia64/kernel/irq_ia64.c
index c94e0bec3d72..8a0171d947e4 100644
--- a/arch/ia64/kernel/irq_ia64.c
+++ b/arch/ia64/kernel/irq_ia64.c
@@ -370,7 +370,7 @@ void destroy_and_reserve_irq(unsigned int irq)
 {
 	unsigned long flags;
 
-	irq_init_desc(irq);
+	irq_free_descs(irq, 1);
 	spin_lock_irqsave(&vector_lock, flags);
 	__clear_irq_vector(irq);
 	irq_status[irq] = IRQ_RSVD;
@@ -403,13 +403,13 @@ int create_irq(void)
  out:
 	spin_unlock_irqrestore(&vector_lock, flags);
 	if (irq >= 0)
-		irq_init_desc(irq);
+		irq_free_descs(irq, 1);
 	return irq;
 }
 
 void destroy_irq(unsigned int irq)
 {
-	irq_init_desc(irq);
+	irq_free_descs(irq, 1);
 	clear_irq_vector(irq);
 }
 
-- 
2.25.1

